src/libsyntax/feature_gate.rs

```diff
@@ -385,6 +386,13 @@ impl<'a> Context<'a> {
                 return;
             }
         }
+        for &(ref n, ref ty) in self.plugin_attributes.iter() {
+            if &*n == name {
+                // Plugins can't gate attributes, so we don't check for it
```

Comment:
The code below doesn't look at `Gated` attributes at all, the only reason for this shortcircuit to exist here would be to avoid the `custom_attribute` feature gate. :)